### DISCUSSION TITLE
navigating using <Link /> not passing `getServerSideProps` to target page

### DISCUSSION THREAD

**Discussion by xta** [vercel/next.js#57310] [Help] [answered]
### Summary

On production vercel hosting, I am seeing `{}` from the page json response when navigating with `<Link>`.  This causes the target page to be loaded without the `getServerSideProps` data.

When navigating directly (putting the target URL in the browser), the `getServerSideProps` are properly setup. 

Oddly, this is only happening on production site (hosted on vercel) and not in other environments (such as localhost). On localhost, I am seeing the correct `getServerSideProps` data object in the Network dev tools tab.

### Additional information

```js
Using next v13.5.6 with the pages router. Pages have server props:


export const getServerSideProps = async () => {
  return {
    props: {
      foo: 'abc',
      bar: 'def',
    },
  };
};

```

Link to looks like this

```
<Link
  href={'/path/goes/here'}
  prefetch={false}
>
  Click me
</Link>

```
```


### Example

See Additional information. Hard to provide full codebase since it may involve the prod environment

**Comment by icyJoseph** [comment]
Do you have anything special going on: middleware, _app, or PWA Service worker?

**Reply by xta** [reply]
using pages/_app with `<Component {...pageProps} />`. It is working (generally works), so I don't think it's an _app issue. 

I don't see any thing of note with regards to middleware or PWA.

**Reply by icyJoseph** [reply]
Is it possible to see this repository? or could you try to share an reproduction? Is there any other route that might be colliding with your route? like catch-all routes maybe?

**Reply by xta** [reply]
Thanks @icyJoseph While I can't share my repo, I am working to reproduce the issue

And if I figure out what's going on, I'll keep this discussion updated

**Reply by xta** [reply]
One thing I noticed in production, the SPA route change is grabbing from disk cache (with empty object response `{}`) when I inspect network tools. On other environments, it's not hitting the cache and grabbing correct server props in response

**Reply by icyJoseph** [reply]
Does that request have a gear icon?

Could you take a screenshot of that? Hide sensitive information.

This sounds like a Service worker issue somehow... Or that maybe your prod app is not getting properly updated as you push changes 🤔

**Reply by xta** [reply]
I don't see a gear icon.

On production deployment:

With `<Link>` mouse hover, I am seeing prefetch that hits CF-Cache (cloudflare)

On `<Link>` click, I'm seeing the fetch event hit disk cache. `Cache-Control: public, max-age=86400; Cf-Cache-Status: HIT`. This is not expected since the destination page is using `getServerSideProps`.

We are using next that includes this (https://github.com/vercel/next.js/blob/v13.5.6/packages/next/src/server/base-server.ts#L1700), but getting `cache-control` public instead of private, no-cache, etc.

**Reply by icyJoseph** [reply]
Is this route using edge runtime?

**Reply by xta** [reply]
no, not using edge runtime

**Reply by icyJoseph** [reply]
Mmm does downgrading next.js versions help? I haven't had time to test this but this looks quite mysterious...

You don't have any constant build-id setup, or some caching layer/configuration? Does the next js config add headers? Maybe a vercel.json file?

**Reply by xta** [reply, marked as answer]
update: cloudflare was caching and breaking the site. it's fixed now
